patches/server/0728-Check-player-world-in-endPortalSoundRadius.patch

```diff
@@ -0,0 +1,20 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: lexikiq <noellekiq@gmail.com>
+Date: Sat, 17 Jul 2021 20:37:02 -0400
+Subject: [PATCH] Check player world in endPortalSoundRadius
+
+Fixes Spigot's endPortalSoundRadius not checking player worlds
+
+diff --git a/src/main/java/net/minecraft/world/item/EnderEyeItem.java b/src/main/java/net/minecraft/world/item/EnderEyeItem.java
+index 7ccfe737fdf7f07b731ea0ff82e897564350705c..42e8f1c269bfbd4f6205ef72ab6026c662551b12 100644
+--- a/src/main/java/net/minecraft/world/item/EnderEyeItem.java
++++ b/src/main/java/net/minecraft/world/item/EnderEyeItem.java
+@@ -66,7 +66,7 @@ public class EnderEyeItem extends Item {
+                         double deltaX = soundPos.getX() - player.getX();
+                         double deltaZ = soundPos.getZ() - player.getZ();
+                         double distanceSquared = deltaX * deltaX + deltaZ * deltaZ;
+-                        if (world.spigotConfig.endPortalSoundRadius > 0 && distanceSquared > world.spigotConfig.endPortalSoundRadius * world.spigotConfig.endPortalSoundRadius) continue; // Spigot
++                        if (world.spigotConfig.endPortalSoundRadius > 0 && (distanceSquared > world.spigotConfig.endPortalSoundRadius * world.spigotConfig.endPortalSoundRadius || !player.getLevel().uuid.equals(world.getMinecraftWorld().uuid))) continue; // Spigot // Paper - ensure recipient is in same world as portal
```

Comment:
You can just check world equality instead of uuids, *perhaps* even referential equality...

Reply:
ServerLevel doesn't actually have an `#equals` method defined, it just falls back to the standard `this == that`. i could use the CraftWorld's `#equals` i guess? it's basically just `this == that || this.uuid.equals(that.uuid)`

Reply:
theres only gonna be one instance of a server level to care about, a player won't be in a server level which isn't the "real" one, comparing UUIDs is fine, most stuff uses reference and falls back to actually checking the values